.github/workflows/sycl_post_commit.yml

```diff
@@ -0,0 +1,93 @@
+name: Linux Post Commit Checks
+
+on:
+  push:
+    branches:
+      - sycl
+jobs:
+  linux_default:
+    name: Linux Default
+    uses: intel/llvm/.github/workflows/sycl_linux_build_and_test.yml@sycl
+    with:
+      build_cache_root: "__w/llvm"
+      build_github_cache: true
+      build_artifact_suffix: default
+  linux_no_assert:
+    name: Linux Default
+    uses: intel/llvm/.github/workflows/sycl_linux_build_and_test.yml@sycl
+    with:
+      build_cache_root: "__w/llvm"
+      build_github_cache: true
+      build_cache_suffix: gcc_no_assertions
+      build_artifact_suffix: gcc_no_assertions
+      build_configure_extra_args: --no-assertions
+  linux_clang_shared_libs:
+    # TODO use nightly images when Docker containers are available
+    name: Linux + Clang + Shared libs
```

Comment:
Can't we add "shared libs" option to sycl_linux_build_and_test workflow?

Reply:
It is possible to pass `--shared-libs` to configure.py with existing configuration, but my intention is to have a nightly compiler image and build inside it. The reason I can't reuse it is sycl_linux_build_and_test workflow doesn't allow to install additional software, which is needed to build this configuration.

Reply:
Technically, we can re-use DPC++ compiler build for this configuration instead of installing clang via apt. Will this simplify the script or we are using "clean" image w/o pre-built DPC++ compiler?

Reply:
> Will this simplify the script or we are using "clean" image w/o pre-built DPC++ compiler?

`sycl_linux_build_and_test` workflow allows you to specify base image, host compiler and additional configure.py flags. Once we have an image with baked in DPC++ compiler, I can leverage all three to switch this job to `sycl_linux_build_and_test` and remove the rest of the code in this file. The intention is to have more "self-compile"-like environment, because we also modify clang frontend, and it would be great to make sure, that it can still compile complex C++ projects.